File: code/python/70_climbing_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        # 本质上是斐波那契数列
        if n <= 2:
            return n
        a, b = 1, 2
        for _ in range(3, n + 1):
            a, b = b, a + b
        return b

File: code/rust/70_climbing_stairs.rs
pub fn climb_stairs(n: i32) -> i32 {
    let mut prev = 1;
    let mut curr = 1;
    for _ in 2..=n {
        let next = prev + curr;
        prev = curr;
        curr = next;
    }
    curr
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_examples() {
        assert_eq!(climb_stairs(2), 2);
        assert_eq!(climb_stairs(3), 3);
        assert_eq!(climb_stairs(4), 5);
        assert_eq!(climb_stairs(5), 8);
    }
}